src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -163,32 +164,36 @@ public void launch(SlaveComputer computer, TaskListener listener) {
                 if (validStates.contains(pod.getStatus().getPhase())) {
                     break;
                 }
-            }
+
+                ++waitedSec;
+            } while (waitedSec < waitForPodSec);
             String status = pod.getStatus().getPhase();
             if (!validStates.contains(status)) {
-                throw new IllegalStateException("Container is not running after " + j + " attempts, status: " + status);
+                throw new IllegalStateException(
+                        "Container is not running after " + waitForPodSec + " seconds, status: " + status);
             }
 
-            j = unwrappedTemplate.getSlaveConnectTimeout();
+            int waitForSlaveToConnect = unwrappedTemplate.getSlaveConnectTimeout();
 
             // now wait for agent to be online
             SlaveComputer slaveComputer = slave.getComputer();
             if (slaveComputer == null) {
                 throw new IllegalStateException("Node was deleted, computer is null");
             }
-            for (; i < j; i++) {
+            for (int waitedForSlave = 0; waitedForSlave < waitForSlaveToConnect; waitedForSlave++) {
                 if (slaveComputer.isOnline()) {
                     break;
                 }
-                LOGGER.log(INFO, "Waiting for agent to connect ({1}/{2}): {0}", new Object[]{podId, i, j});
-                logger.printf("Waiting for agent to connect (%2$s/%3$s): %1$s%n", podId, i, j);
+                LOGGER.log(INFO, "Waiting for agent to connect ({1}/{2}): {0}",
+                        new Object[]{podId, waitedForSlave, waitForSlaveToConnect});
+                logger.printf("Waiting for agent to connect (%2$s/%3$s): %1$s%n",
+                        podId, waitedForSlave, waitForSlaveToConnect);
                 Thread.sleep(1000);
             }
             if (!slaveComputer.isOnline()) {
-                if (containerStatuses != null) {
-                    logLastLines(containerStatuses, podId, namespace, slave, null, client);
-                }
-                throw new IllegalStateException("Agent is not connected after " + j + " attempts, status: " + status);
+                logLastLines(containerStatuses, podId, namespace, slave, null, client);
```

Comment:
why remove the null check ?

Reply:
Since we refer to `containerStatuses` in [line 133](https://github.com/jenkinsci/kubernetes-plugin/pull/398/files/b4a83589f24e36b94f5ceb1510080fd9e9837af0#diff-d31a23c33d2739a785384f53a13bc682R133), there's no way `containerStatuses` will be null at this point. Otherwise, we would have got into a NPE earlier.

findbugs found this error for me at one of my earlier commits in this PR.